src/identity-utils.ts

```diff
@@ -0,0 +1,201 @@
+import Constants from './constants';
+import { Dictionary, parseNumber } from './utils';
+import { BaseVault } from './vault';
+import Types from './types';
+import { IdentityApiData, UserIdentities } from '@mparticle/web-sdk';
+import { IdentityAPIMethod } from './sdkRuntimeModels';
+import { isObject } from './utils';
+const { Identify, Modify, Login, Logout } = Constants.IdentityMethods;
+import { ONE_DAY_IN_SECONDS, MILLIS_IN_ONE_SEC } from './constants';
+import { generateHash } from './utils';
+
+export interface IKnownIdentities extends UserIdentities {
+    device_application_stamp?: string;
+}
+
+export interface ICachedIdentityCall {
+    responseText: string;
+    status: number;
+    expireTimestamp: number;
+}
+
+export const cacheOrClearIdCache = (
+    method: string,
+    knownIdentities: IKnownIdentities,
+    idCache: BaseVault<Dictionary<ICachedIdentityCall>>,
+    xhr: XMLHttpRequest
+): void => {
+    const CACHE_HEADER = 'x-mp-max-age';
```

Comment:
I think this should go to `./constants.js`. I can see us reusing this elsewhere.

Reply:
I'm going to push back on this one. I actually don't see a use right now for the future.  If there is though then I can move it over during that code commit when it is used elsewhere, but since it's just used in this function, I'd prefer to keep it here.